packages/mobile/src/screens/edit-track-screen/fields/VisibilityField/VisibilityScreen.tsx

```diff
@@ -0,0 +1,162 @@
+          is_scheduled_release: false,
+          release_date: null
+        })
+        break
+      case 'hidden':
+        setValues({
+          ...values,
+          is_unlisted: true,
+          is_scheduled_release: false
+        })
+        break
+      case 'scheduled':
+        if (!releaseDate) {
+          setDateError('Release date required')
+          return
+        } else if (dayjs(releaseDate).isBefore(dayjs())) {
+          setDateTimeError('Release date must be in the future')
+          return
+        } else {
+          setValues({
+            ...values,
+            is_unlisted: true,
+            is_scheduled_release: true,
+            release_date: releaseDate
+          })
+        }
+        break
+    }
+    navigation.goBack()
+  }, [visibilityType, releaseDate, navigation, values, setValues])
+
+  return (
+    <FormScreen
+      title={messages.title}
+      icon={IconVisibilityPublic}
+      onSubmit={handleSubmit}
+      stopNavigation
+    >
+      <ExpandableRadioGroup
+        value={visibilityType}
+        onValueChange={setVisibilityType}
+      >
+        <ExpandableRadio
+          value='public'
+          label={messages.public}
+          description={messages.publicDescription}
+        />
+        <ExpandableRadio
+          value='hidden'
+          label={messages.hidden}
+          icon={IconVisibilityHidden}
+          description={messages.hiddenDescription}
+          disabled={isAlreadyPublic}
+        />
+        <ExpandableRadio
+          value='scheduled'
+          label={messages.scheduledRelease}
+          description={messages.scheduledReleaseDescription}
+          disabled={isAlreadyPublic}
+          checkedContent={
+            <Flex direction='column' gap='l'>
+              <Flex gap='l'>
+                <Flex gap='l'>
+                  <DateTimeInput
+                    mode='date'
+                    date={releaseDate ?? undefined}
+                    onChange={setReleaseDate}
+                    inputProps={{
+                      label: messages.dateLabel,
+                      startIcon: IconCalendarMonth,
+                      error: !!(dateError || dateTimeError),
+                      helperText: dateError
+                    }}
+                    dateTimeProps={{
+                      minimumDate: dayjs().toDate()
+                    }}
+                  />
+                  <DateTimeInput
+                    mode='time'
+                    date={releaseDate ?? undefined}
+                    onChange={setReleaseDate}
+                    inputProps={{
+                      label: messages.timeLabel,
+                      error: !!dateTimeError,
+                      helperText: dateTimeError
+                    }}
+                  />
+                </Flex>
+              </Flex>
+              {releaseDate ? (
+                <Hint>{messages.futureReleaseHint(getLocalTimezone())}</Hint>
+              ) : null}
+            </Flex>
```

Comment:
I'd be tempted to factor this out

Reply:
so i actually did, and then felt the level of indirection was kinda annoying so i brought it back :D

Reply:
basically was passing through 5+ props